src/d3-graph.ts

```diff
@@ -1,53 +1,56 @@
 /**
  * TODO
  * - Keep track of transformation to graph (translation and scale) when rerendering
+ * - Use correct D3 types instead of `any`
  */
 
 import * as d3 from 'd3'
 import * as d3zoom from 'd3-zoom'
 
 import {getLogger} from './logger'
+import {Dimensions, El, GraphData} from './types'
 
 const logger = getLogger('d3-graph')
 
 export default class D3Graph {
 
-  constructor() {
-    this._LABEL_FONT_SIZE = 8
-    this._TRANSITION_DURATION = 250
-    this._PAN_MOVEMENT_OFFSET = 50
-
-    document.d3Initialized = false
-    this._width = null
-    this._height = null
-    this._host = null
-    this._svg = null
-    this._c10 = null
-    this._g = null
-    this._zoomHandler = null
-
-    this._links = null
-    this._nodes = null
-    this._drag = null
-    this._nodeTextLabels = null
-
-    this._selectedNode = null
-    this._mouseDownNode = null
+  private static readonly _LABEL_FONT_SIZE = 8
+  private static readonly _TRANSITION_DURATION = 250
+  private static readonly _PAN_MOVEMENT_OFFSET = 50
+
+  private _width: number | null = null
+  private _height: number | null = null
+  private _host: El | null = null
+  private _svg: any | null = null
+  private _c10: any | null = null
+  private _g: any | null = null
+  private _zoomHandler: any| null = null
+
+  private _links: any | null = null
+  private _nodes: any | null = null
+  private _drag: any | null = null
+  private _nodeTextLabels: any| null = null
+
+  private _selectedNode: any | null = null
+  private _mouseDownNode: any | null = null
+
+  public constructor() {
+    (document as any).d3Initialized = false
   }
 
-  init(host, dimensions) {
+  public init(host: El, dimensions: Dimensions): void {
```

Comment:
A comment should be added here to describe this function, similar to render() and _renderNodes() below.

Reply:
I added those before I converted to typescript just for personal documentation. But in general documentation on functions is needed. you can open a new issue for this